.github/workflows/release-with-changesets.yml

```
@@ -0,0 +1,137 @@
+        id: lockversion
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "${{ steps.lockversion.outputs.version }}"
+      - if: steps.lockversion.outputs.version == '18' && matrix.os == 'windows-latest'
+        name: Install npm cli 8
+        shell: bash
+        # npm cli 10 is buggy because of some cache issues
+        run: npm install -g npm@8.19.4
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Install dependencies
+        shell: bash
+        run: npm ci
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Run test
+        run: npm test --if-present
+      - if: failure() # Only, on failure, send a message on the 94_bot-failing-ci slack channel
+        name: Report workflow run status to Slack
+        uses: 8398a7/action-slack@v3
+        with:
+          status: ${{ job.status }}
+          fields: repo,action,workflow
+          text: "Release workflow failed in testing job"
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_CI_FAIL_NOTIFY }}
+
+  release:
+    needs: [test-nodejs]
+    name: Publish to any of NPM, GitHub, or Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set git to use LF # To once and for all finish the never-ending fight between Unix and Windows
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Check if Node.js project and has package.json
+        id: packagejson
+        run: test -e ./package.json && echo "exists=true" >> $GITHUB_OUTPUT || echo "exists=false" >> $GITHUB_OUTPUT
+        shell: bash
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Check package-lock version
+        uses: asyncapi/.github/.github/actions/get-node-version-from-package-lock@master
+        id: lockversion
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "${{ steps.lockversion.outputs.version }}"
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Install dependencies
+        shell: bash
+        run: npm ci
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Install changelog
+        shell: bash
+        # This step can be removed once the issue is fixed in the changeset package.
+        run: npm install @changesets/changelog-git@0.2.0
+      - if: steps.packagejson.outputs.exists == 'true'
+        name: Publish to any of NPM, Github, and Docker Hub
+        uses: changesets/action@v1
+        id: release
+        with:
+          version: npx -p @changesets/cli changeset version
+          commit: "feat: version packages"
+          title: "feat: version packages"
```

```suggestion
          commit: "chore(release): release and bump versions of packages"
          title: "chore(release): release and bump versions of packages"
```

sorry, update also readme pleasae

now since I know how release will work, I understand why initially it was `chore`, lets just add more details

---

ok done.